packages/react-components/src/components/ChatMessage/ChatMessageActionsFlyout.tsx

```diff
@@ -32,34 +41,93 @@ export interface ChatMessageActionFlyoutProps {
  * @private
  */
 export const ChatMessageActionFlyout = (props: ChatMessageActionFlyoutProps): JSX.Element => {
-  const menuItems = useMemo(
-    (): IContextualMenuItem[] => [
-      {
-        key: 'Edit',
-        text: props.strings.editMessage,
-        itemProps: { styles: props.increaseFlyoutItemSize ? menuItemIncreasedSizeStyles : undefined },
-        iconProps: { iconName: 'MessageEdit', styles: menuIconStyleSet },
-        onClick: props.onEditClick
-      },
-      {
-        key: 'Remove',
-        text: props.strings.removeMessage,
-        itemProps: { styles: props.increaseFlyoutItemSize ? menuItemIncreasedSizeStyles : undefined },
-        iconProps: {
-          iconName: 'MessageRemove',
-          styles: menuIconStyleSet
+  const theme = useTheme();
+  const menuItems = useMemo((): IContextualMenuItem[] => {
+    // only show read by x of x if more than 3 participants in total including myself
+    if (
+      props.participantCountNotIncludingSelf &&
+      props.messageThreadReadCount !== undefined &&
+      props.participantCountNotIncludingSelf >= 2 &&
+      props.strings.messageReadCount
+    ) {
+      return [
+        {
+          key: 'Edit',
```

Comment:
Don't duplicate the menu items for "edit", and "remove" in the two branches. Instead, first create the array with those two, then add in the read count related items if the conditions are met.